clustered/client/src/main/java/org/ehcache/clustered/client/config/TimeoutDuration.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.config;
+
+import java.math.BigInteger;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Describes a timeout value for a clustered operation.  Instances of this class
+ * are subject to the operation of the Java {@code java.util.concurrent.TimeUnit} class.
+ */
+public final class TimeoutDuration {
+  private final long amount;
+  private final TimeUnit unit;
+
+  private TimeoutDuration(long amount, TimeUnit unit) {
+    this.amount = amount;
+    this.unit = unit;
+  }
+
+  /**
+   * Constant indicating no timeout.
+   */
+  public static final TimeoutDuration NONE = new TimeoutDuration(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
```

Comment:
This currently means that I can define legal durations longer than "NONE", isn't that a bit strange?